controllers/bucket_controller.go

```diff
@@ -375,6 +377,8 @@ func (r *BucketReconciler) reconcileStorage(ctx context.Context, obj *sourcev1.B
 	if artifact := obj.GetArtifact(); artifact != nil && !r.Storage.ArtifactExist(*artifact) {
 		obj.Status.Artifact = nil
 		obj.Status.URL = ""
+		// Remove the condition as the artifact doesn't exist.
+		conditions.Delete(obj, sourcev1.ArtifactInStorageCondition)
 	}
```

Comment:
Any chance of `artifact` being `nil` and we still want to delete the condition?

```go
if artifact := obj.GetArtifact(); artifact == nil || !r.Storage.ArtifactExist(*artifact) {
```

Reply:
Yes, that can also be added in the following block along with setting reconciling condition.

Reply:
Added to all the reconcilers.